core/templates/pages/skill-editor-page/editor-tab/skill-misconceptions-editor/misconception-editor.directive.spec.ts

```diff
@@ -0,0 +1,173 @@
+
+    directive = $injector.get('misconceptionEditorDirective')[0];
+    skillEditorStateService = $injector.get('SkillEditorStateService');
+    skillUpdateService = $injector.get('SkillUpdateService');
+    skillObjectFactory = $injector.get('SkillObjectFactory');
+    misconceptionObjectFactory = $injector.get('MisconceptionObjectFactory');
+
+    sampleSkill = skillObjectFactory.createInterstitialSkill();
+    sampleMisconception = misconceptionObjectFactory.create(
+      'misconceptionId', 'name', 'notes', 'feedback', false);
+
+    spyOn(skillEditorStateService, 'getSkill').and.returnValue(sampleSkill);
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope,
+      $scope: $scope
+    });
+    $scope.misconception = sampleMisconception;
+    $scope.isEditable = function() {
+      return true;
+    };
+    ctrl.$onInit();
+  }));
+
+  it('should set properties when initialized', function() {
+    expect($scope.nameEditorIsOpen).toEqual(false);
+    expect($scope.notesEditorIsOpen).toEqual(false);
+    expect($scope.feedbackEditorIsOpen).toEqual(false);
+    expect($scope.skill).toEqual(sampleSkill);
+  });
+
+  it('should open name editor when clicking on edit button', function() {
+    expect($scope.nameEditorIsOpen).toBe(false);
+    $scope.openNameEditor();
+    expect($scope.nameEditorIsOpen).toBe(true);
+  });
+
+  it('should open notes editor when clicking on edit button', function() {
+    expect($scope.notesEditorIsOpen).toBe(false);
+    $scope.openNotesEditor();
+    expect($scope.notesEditorIsOpen).toBe(true);
+  });
+
+  it('should open feedback editor when clicking on edit button', function() {
+    expect($scope.feedbackEditorIsOpen).toBe(false);
+    $scope.openFeedbackEditor();
+    expect($scope.feedbackEditorIsOpen).toBe(true);
+  });
+
+  it('should save name when clicking on save button', function() {
+    let updateNameSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionName').and.returnValue(null);
+
+    $scope.openNameEditor();
+    // Setting new name.
+    $scope.container.misconceptionName = 'newName';
+    $scope.saveName();
+
+    expect(updateNameSpy).toHaveBeenCalled();
+  });
+
+  it('should save notes when clicking on save button', function() {
+    let updateNotesSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionNotes').and.returnValue(null);
+
+    $scope.openNotesEditor();
+    // Setting new notes content.
+    $scope.container.misconceptionNotes = 'newNotes';
+    $scope.saveNotes();
+
+    expect(updateNotesSpy).toHaveBeenCalled();
+  });
+
+  it('should save feedback when clicking on save button', function() {
+    let updateFeedbackSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionFeedback').and.returnValue(null);
+
+    $scope.openFeedbackEditor();
+    // Setting new feedback content.
+    $scope.container.misconceptionFeedback = 'newFeedback';
+    $scope.saveFeedback();
+
+    expect(updateFeedbackSpy).toHaveBeenCalled();
```

Comment:
Is it possible to check what it was called with?

Reply:
done